lib/samples/filter_by_definition_expression_or_display_filter/filter_by_definition_expression_or_display_filter_sample.dart

```diff
@@ -82,24 +91,61 @@ class _FilterByDefinitionExpressionOrDisplayFilterSampleState
   }
 
   void applyDefinitionExpression() {
+    // remove the display filter
     _featureLayer.displayFilterDefinition = null;
+    // apply a definition expression to the feature layer
     _featureLayer.definitionExpression =
         "req_Type = 'Tree Maintenance or Damage'";
+    // count the number of features
+    calculateFeatureCount();
   }
 
   void applyDisplayFilter() {
+    // remove the definition expression
     _featureLayer.definitionExpression = '';
+    // apply a display filter to the feature layer
     final displayFilter = DisplayFilter.withWhereClause(
         name: 'Damaged Trees',
         whereClause: "req_type LIKE '%Tree Maintenance%'");
+    // create a manual display filter definition
     final manualDisplayFilterDefinition =
         ManualDisplayFilterDefinition.withFilters(
             activeFilter: displayFilter, availableFilters: [displayFilter]);
     _featureLayer.displayFilterDefinition = manualDisplayFilterDefinition;
+    // count the number of features
+    calculateFeatureCount();
   }
 
   void reset() {
+    // remove the definition expression and display filter
     _featureLayer.displayFilterDefinition = null;
     _featureLayer.definitionExpression = '';
+    // count the number of features
+    calculateFeatureCount();
+  }
+
+  void calculateFeatureCount() async {
+    // count the number of features
+    final queryParameters = QueryParameters();
+    // get the current extent of the map view
+    queryParameters.geometry = _mapViewController
+        .getCurrentViewpoint(viewpointType: ViewpointType.boundingGeometry)
+        ?.targetGeometry
+        .extent;
+    queryParameters.whereClause = '1=1';
+
+    // query the feature count
+    final featureCount = await _featureLayer.featureTable!
+        .queryFeatureCount(queryParameters: queryParameters);
+
+    // show the feature count in an alert dialog
+    if (mounted) {
+      showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(content: Text('$featureCount features found.'));
```

Comment:
Let's have a Title 'Current feature count' and Content as `$featureCount features`. This aligns with the common design and iOS implementation, and makes the dialog look a little tidier, since we will be keeping the dialog for this sample and not looking to implement a callout or any other UI alternative.

Reply:
Wait @jenmerritt will we be in trouble for not following the commom design spec where the featureCount is just part of the bottom bar as plain text. When I was referring to iOS Sample and they have an alert dialog so I though that is more appropriate for a mob app.

Reply:
No the common design is desktop, we've implemented a better UI for mobile so this is fine in my opinion. The sample is emphasising the difference between a display filter and definition expression so I think showing the results in the dialog is a good pattern to follow.

Reply:
Okay, thanks!